Python/Product/Uap/Project/PythonUapProjectConfig.cs

```diff
@@ -0,0 +1,961 @@
+                appPackageDebugTarget[0].bstrCurDir = targets[0].bstrCurDir;
+                appPackageDebugTarget[0].bstrEnv = targets[0].bstrEnv;
+                appPackageDebugTarget[0].dwProcessId = targets[0].dwProcessId;
+                appPackageDebugTarget[0].pStartupInfo = IntPtr.Zero;
+                appPackageDebugTarget[0].guidLaunchDebugEngine = targets[0].guidLaunchDebugEngine;
+                appPackageDebugTarget[0].dwDebugEngineCount = targets[0].dwDebugEngineCount;
+                appPackageDebugTarget[0].pDebugEngines = targets[0].pDebugEngines;
+                appPackageDebugTarget[0].guidPortSupplier = targets[0].guidPortSupplier;
+
+                appPackageDebugTarget[0].bstrPortName = targets[0].bstrPortName;
+                appPackageDebugTarget[0].bstrOptions = targets[0].bstrOptions;
+                appPackageDebugTarget[0].fSendToOutputWindow = targets[0].fSendToOutputWindow;
+                appPackageDebugTarget[0].pUnknown = targets[0].pUnknown;
+                appPackageDebugTarget[0].guidProcessLanguage = targets[0].guidProcessLanguage;
+                appPackageDebugTarget[0].project = PythonConfig;
+
+                // Pass the debug launch targets to the debugger
+                IVsDebugger4 debugger4 = (IVsDebugger4)Package.GetGlobalService(typeof(SVsShellDebugger));
+
+                VsDebugTargetProcessInfo[] results = new VsDebugTargetProcessInfo[1];
+
+                IVsDebugger debugger = (IVsDebugger)debugger4;
+
+                // Launch task to monitor to attach to Python remote process
+                var sourceDir = System.IO.Path.GetFullPath(PythonConfig.GetProjectProperty("ProjectDir")).Trim('\\');
+                var targetDir = System.IO.Path.GetFullPath(this.LayoutDir).Trim('\\');
+                var debugPort = "5678";
+                var debugId = Guid.NewGuid();
+                var debugXml = new XDocument(new XElement("dbg",
+                    new XElement("arg", @"visualstudio_py_remote_launcher.py"),
+                    new XElement("arg", debugPort),
+                    new XElement("arg", debugId),
+                    new XElement("arg", "--redirect-output")));
+
+                Debugger.PythonRemoteDebugEvents.Instance.AttachRemoteDebugXml = debugXml.ToString();
+                Debugger.PythonRemoteDebugEvents.Instance.AttachRemoteProcessFunction = () => {
+                    return RemoteProcessAttachAsync(
+                         appPackageDebugTarget[0].bstrRemoteMachine,
+                         debugId.ToString(),
+                         debugPort,
+                         sourceDir,
+                         targetDir);
+                };
+
+                int result = debugger.AdviseDebugEventCallback(Debugger.PythonRemoteDebugEvents.Instance);
+                System.Diagnostics.Debug.Assert(result == VSConstants.S_OK, string.Format(System.Globalization.CultureInfo.CurrentCulture, "Failure {0}", result));
```

Comment:
What happens if this fails in a release build? We should handle the error case somehow (even throwing an exception that will trigger error reporting is better than silently trying to continue, probably into a deadlock).